verifcid/allowlist.go

```diff
@@ -0,0 +1,58 @@
+package verifcid
+
+import (
+	"github.com/ipfs/go-cid"
+	mh "github.com/multiformats/go-multihash"
+)
+
+// Allowlist defines an interface containing list of allowed multihashes.
+type Allowlist interface {
```

Comment:
I feel like it should be moved to go-cid instead. Although, it's not that widely used

Reply:
I remember discussing with @aschmahmann about it, he said that back then go-verifcid was supposed to move into go-cid but this didn't happend. This would have positives, very few pieces of code and peoples are aware they probably want to use this package.

I don't know if we want to move it now, it does not really help with the globals problem we are having here.

Reply:
As we removed ValidateCID, the Allowlist would better be moved to go-multihash